lib/src/component/with_transition.dart

```diff
@@ -0,0 +1,426 @@
+
+  @override
+  get defaultProps => (newProps()
+    ..addProps(v2.TransitionPropsMixin.defaultProps)
+    ..isShown = false
+    ..transitionTimeout = const Duration(seconds: 1)
+    ..childPropsByPhase = const {
+      v2.TransitionPhase.PRE_SHOWING: {},
+      v2.TransitionPhase.SHOWING: {},
+      v2.TransitionPhase.SHOWN: {},
+      v2.TransitionPhase.HIDING: {},
+      v2.TransitionPhase.HIDDEN: {},
+    }
+  );
+
+  @override
+  get initialState => (newState()
+    ..$transitionPhase = _transitionPhaseDerivedFromProps(props)
+  );
+
+  @override
+  getDerivedStateFromProps(Map nextProps, Map prevState) {
+    final tNextProps = typedPropsFactory(nextProps);
+    final tPrevState = typedStateFactory(prevState);
+    if (tNextProps.isShown && _isOrWillBeHidden(tPrevState.$transitionPhase)) {
+      return _stateToBeginShowing(tNextProps, tPrevState.$transitionPhase);
+    } else if (!tNextProps.isShown && _isOrWillBeShown(tPrevState.$transitionPhase)) {
+      return _stateToBeginHiding(tNextProps, tPrevState.$transitionPhase);
+    }
+
+    return null;
+  }
+
+  @override
+  void componentDidUpdate(Map prevProps, Map prevState, [dynamic snapshot]) {
+    if (state.$transitionPhase == typedStateFactory(prevState).$transitionPhase) return;
+
+    _transitionNotGuaranteed = false;
+
+    if (state.$transitionPhase != TransitionPhase.SHOWING) {
+      // Allows the `WithTransitionComponent` to handle state changes that interrupt state
+      // changes waiting on `transitionend` events.
+      _cancelTransitionEventListener();
+    }
+
+    switch (state.$transitionPhase) {
+      case TransitionPhase.PRE_SHOWING:
+        _handlePreShowing();
+        break;
+      case TransitionPhase.SHOWING:
+        // NOTE: props.onWillShow is called within _handleShow in the handler that
+        // triggers an update to `$transitionPhase` that will result in this effect being processed.
+        break;
+      case TransitionPhase.HIDING:
+        // NOTE: props.onWillHide is called within _handleHide in the handler that
+        // triggers an update to `$transitionPhase` that will result in this effect being processed.
+        _transitionNotGuaranteed = state.$transitionPhase == TransitionPhase.SHOWING;
+        _handleHiding();
+        break;
+      case TransitionPhase.HIDDEN:
+        props.onDidHide?.call();
+        break;
+      case TransitionPhase.SHOWN:
+        props.onDidShow?.call();
+        break;
+    }
+  }
+
+  var _isMounted = false;
+
+  @override
+  void componentDidMount() {
+    _isMounted = true;
+  }
+
+  @override
+  void componentWillUnmount() {
+    super.componentWillUnmount();
+
+    _isMounted = false;
+    _cancelTransitionEndTimer();
+    _cancelTransitionEventListener();
+  }
+
+  @override
+  render() {
+    assert(_hasSingleValidChild(props));
+
+    final childElement = props.children.single as ReactElement;
+    final childProps = domProps({...getProps(childElement)});
```

Comment:
#nit unnecessary map copy
```suggestion
    final childProps = domProps(getProps(childElement));
```